modules/core/shared/src/main/scala/gem/optics/README.md

```diff
@@ -0,0 +1,32 @@
+## Non-Injective Optics
+
+Standard 2-way optics deal with **invertible** mappings. `Iso[A, B]` says that `A` and `B` are equal, so round-trips in either direction are identities. `Prism[A, B]` says that there is some *subset* of `A` that is equal to `B`.
+
+If we loosen the requirement that types be the same size we get a different kind of mapping, where the large type is squeezed into the small type in one direction or the other. An example is `Int ⟺ Byte` by the standard widening/narrowing conversions. Note that the round-trip starting at `Byte` is an identity, but the round-up starting at `Int` is merely **idempotent**: the first round-trip "normalizes" an `Int` into `Byte` range and thereafter the round-trip is an identity.
+
+This [phenomenon](https://ncatlab.org/nlab/show/split+epimorphism) is a thing, called a **split monomorphism** or a **split epimorphism** depending on which side is bigger. Note that every `Iso` is trivially a split where the idempotent round-trip happens to be an identity.
+
+When we compose a split mono and a split epi end-to-end in either direction we end up with a situation where neither round-trip is necessarily an identity but both are idempotent. I'm calling this a `Wedge` for lack of a better idea. Splits are trivially wedges where one of the idempotent round-trips happens to be an identity.
+
+A `Format` is a weaker `Prism` where a *subset* of `A` forms a split epi with `B`. Every `Prism` is a `Format` where the split epi happens to be an `Iso`; and every `SplitEpi` forms a `Prism` where the subset of `A` is `A` itself.
+
+
+```
+               Wedge[A,B]
+                 A ? B
+
+                   │                  Format[A,B]
+          ┌────────┴────────┐       ∃ a ⊂ A | a > B
+          │                 │
+                                           │
+    SplitMono[A,B]     SplitEpi[A,B]  ─────┤
+        A < B             A > B            │
+
+          │                 │         Prism[A,B]
+          └────────┬────────┘       ∃ a ⊂ A | a = B
+                   │                       │
+                                           │
+                Iso[A,B]  ─────────────────┘
+                 A = B
```

Comment:
This was all very clearly written and understandable (thank you for making this easily digestible). My only confusion is with the use of the symbols `<` and `>`.

For a monomorphism, isn't it required that `A` map surjectively onto `B` (hence I'm not sure what `A < B` means, as `|A| ≥ |B|` is necessary to be able to be surjective), and for an epimorphism `φ`, isn't it required that for all `x, y` in `A`, `x ≠ y` implies `φ(x) ≠ φ(y)`, i.e. the mapping is injective? (And in that case, `|A| ≤ |B|` is required.)

Furthermore, just a pedantic point of mathematical contention, but an isomorphism (or other mapping) between `A` and `B` doesn't really mean that `A = B`: it means that there is an invertible bijective map `φ` such that `φ(A) = B`.

Reply:
I may misunderstand what you're saying … monomorphisms generalize injective functions, epimorphisms generalize surjective functions. I'm following the definitions [here](https://ncatlab.org/nlab/show/split+monomorphism) where `get` is the mono/epi and `reverseGet` is its retraction/section. 

And yeah, the '<', '>', '=' thing is just an intuition.

Reply:
Ahhh, you're right. I haven't studied this in years and now I get to suffer the embarrassment of having mixed up epimorphisms and monomorphisms! Apologies for that.

Now that my definitions are not reversed, this makes much more sense.

Reply:
No worries. It's really easy to get everything backwards because they're duals. I confused myself several time writing this.